app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsConfirmAddressController.scala

```diff
@@ -58,14 +57,19 @@ class ContactDetailsConfirmAddressController @Inject() (
         case Some(journeyId) =>
           request.registration.organisationDetails.organisationType match {
             case Some(UK_COMPANY) =>
-              executeAddressUpdate(
-                journeyId,
-                journeyId =>
-                  incorpIdConnector.getDetails(journeyId)
-                    .flatMap(
-                      response =>
-                        updateBusinessAddress(response.companyAddress.toPptAddress)(request)
-                    )
+              executeAddressUpdate(journeyId,
+                                   _ =>
+                                     request.registration.organisationDetails.incorporationDetails match {
+                                       case Some(details) =>
+                                         updateBusinessAddress(details.companyAddress.toPptAddress)
+                                       case None =>
+                                         throw DownstreamServiceError(
+                                           "Failed to fetch incorporated details from cache",
+                                           RegistrationException(
+                                             "Failed to fetch incorporated details from cache"
+                                           )
+                                         )
+                                     }
```

Comment:
I think we should take the time to refactor this method here: 
- I don't think journeyId is used any more?
- Can we extract these anonymous functions into helper methods to improve readability?

Reply:
removed the journey id and extracted function to a separate method.